apps/io/search_ingest.py

```diff
@@ -106,3 +110,89 @@ def _get_query(self, req):
         args = getattr(req, "args", {})
         query = json.loads(args.get("source")) if args.get("source") else {"query": {"filtered": {}}}
         return query
+
+
+class SearchIngestServiceAsync(AsyncBaseService):
+    def __init__(self, datasource=None, backend=None, source=None):
+        super().__init__(datasource, backend)
+        self.source = source
+
+    async def get_provider_async(self):
+        resource_service = get_resource_service("search_providers")
+        assert resource_service is not None
+        resource_service = cast(SearchProviderServiceAsync, resource_service)
+        assert resource_service.is_async
+        provider = await resource_service.find_one_async(source=self.source, req=None)
+        if provider and "config" in provider and "username" in provider["config"]:
+            assert self.backend is not None
+            # TODO-ASYNC[search_ingest]: check this method
+            await self.backend.set_credentials_async(provider["config"]["username"], provider["config"].get("password", ""))
+        return provider
+
+    async def fetch_async(self, guid):
+        assert self.backend is not None
+        return self.backend.find_one_raw_async(guid, guid)
+
+    async def create_async(self, docs, **kwargs):
+        new_guids = []
+        provider = await self.get_provider_async()
+        for doc in docs:
+            if not doc.get("desk"):
+                # if no desk is selected then it is bad request
+                raise SuperdeskApiError.badRequestError(_("Destination desk cannot be empty."))
+            try:
+                archived_doc = await self.fetch_async(doc["guid"])
+            except FileNotFoundError as ex:
+                raise ProviderError.externalProviderError(ex, provider)
+
+            # TODO-ASYNC: fetch_item is not async
+            dest_doc = fetch_item(archived_doc, doc.get("desk"), doc.get("stage"), state=doc.get("state"))
```

Comment:
[fetch_item](https://github.com/superdesk/superdesk-core/blob/async/apps/archive/common.py#L467) is now async. Might be good to pull the latest, fix merge conflicts